src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -918,7 +919,25 @@ protected virtual ImmutableArray<AdditionalTextFile> ResolveAdditionalFilesFromA
                 builder.Add(new AdditionalTextFile(file, this));
             }
 
-            return builder.ToImmutableArray();
+            return builder.ToImmutable();
+        }
+
+        protected SyntaxTreeOptionsProvider CreateSyntaxTreeOptionsProvider()
+        {
+            // If we don't have any editor config files, we can opt out entirely
+            if (Arguments.EditorConfigFiles.IsDefaultOrEmpty)
+            {
+                return null;
+            }
+
+            var builder = ImmutableArray.CreateBuilder<AdditionalText>();
```

Comment:
Believe you can use `Arguments.EditorConfigFiles` to provide an accurate count here to the builder.